solr/core/src/java/org/apache/solr/schema/CurrencyFieldType.java

```diff
@@ -387,7 +386,7 @@ private Query getRangeQueryInternal(
         (p1 != null) ? p1.getCurrencyCode() : (p2 != null) ? p2.getCurrencyCode() : defaultCurrency;
 
     // ValueSourceRangeFilter doesn't check exists(), so we have to
-    final Query docsWithValues = new DocValuesFieldExistsQuery(getAmountField(field).getName());
+    final Query docsWithValues = getAmountField(field).getType().getExistenceQuery(parser, field);
```

Comment:
Able to reproduce the `./gradlew test --tests CurrencyFieldTypeTest` test failure locally, shall we go back to `FieldExistsQuery` here maybe, or perhaps the test needs revising (haven't looked).
```suggestion
    final Query docsWithValues = new FieldExistsQuery(getAmountField(field).getName());
```

Reply:
Just debugged this, stupid error. Should be passing the amountField into the method instead of the currencyField.